src/refract/parseResult.es6

```diff
@@ -0,0 +1,48 @@
+/*
+ * Parse result-specific refract elements. Includes the API namespace.
+ * General structure:
+ *
+ * + ParseResult
+ *   + Annotation
+ */
+
+import {
+  ArrayElement, StringElement, registry
+} from 'minim';
+
+import './api';
+
+export class ParseResult extends ArrayElement {
+  constructor(...args) {
+    super(...args);
+    this.element = 'parseResult';
+  }
+
+  get api() {
+    return this.findByClass('api').first();
+  }
+
+  get annotations() {
+    return this.findByElement('annotation');
+  }
+}
+
+export class Annotation extends StringElement {
+  constructor(...args) {
+    super(...args);
+    this.element = 'annotation';
+  }
+
+  get code() {
+    return this.attributes.getValue('code');
+  }
+
+  set code(value) {
+    this.attributes.set('code', value);
+  }
```

Comment:
@zdne yes this makes sense. Since it is not refracted it won't be getting a custom element implementation in Fury for the time being. We only support refracted attributes as elements for now and honestly this is just an array. It'll be loaded as an ArrayElement and you'll get all the niceties exposed through that class.

Also, I seem to have run into an interesting issue where modifying the original class in a hierarchy doesn't update the dependents, so I'm not sure how to add a nice `sourceMap` getter for every defined class. We'll have to revisit that in the future (possibly by updated everything in the registry). For now `element.attributes.get('sourceMap')` will work. Any convenience layer we add will be done in an additive, backward compatible way.

@pksunkara given the above, are you okay with merging this as-is?